NPM automation token

NPM introduced automation tokens https://github.blog/changelog/2020-10-02-npm-automation-tokens/

This means we can keep our 2FA for login and publish and use automation tokens to bypass 2FA for publishing.

I did that on a test project and `semantic-release/npm` fails at the verify auth step for npm.

I was wondering it this is miss configuration on my part if there is work that needs to be done to support the new tokens.

The setup is simple, I am using 2FA for both publish and login on NPM

![image](https://user-images.githubusercontent.com/46779/95470074-426b8400-0978-11eb-89b3-09cd058acd95.png)

and my test repo has enabled the automation token option from settings.

![image](https://user-images.githubusercontent.com/46779/95469963-1f40d480-0978-11eb-9076-4f43172c6cf5.png)

As for my semantic-release config

```
{
	"branches": ["+([0-9])?(.{+([0-9]),x}).x", "main", "next", "next-major", { "name": "beta", "prerelease": true }, { "name": "alpha", "prerelease": true }],
	"plugins": [
		"@semantic-release/commit-analyzer",
		"@semantic-release/release-notes-generator",
		[
			"@semantic-release/changelog",
			{
				"changelogTitle": "# Changelog\n\nAll notable changes to this project will be documented in this file. See\n[Conventional Commits](https://conventionalcommits.org) for commit guidelines."
			}
		],
		[
			"@semantic-release/npm",
			{
				"tarballDir": "pack"
			}
		],
		[
			"@semantic-release/github",
			{
				"assets": "pack/*.tgz"
			}
		],
		[
			"@semantic-release/git",
			{
				"message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
			}
		]
	]
}
```

So it looks like if I use `npm whoami` (and I am successfully logged in via npm login) it works ok.

If I use the generated whoami from the lib:

```
npm whoami --userconfig /private/var/folders/jj/temppathhere/.npmrc --registry https://registry.npmjs.org
npm ERR! code E403
npm ERR! 403 403 Forbidden - GET https://registry.npmjs.org/-/whoami
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/radu.micu/.npm/_logs/2020-10-08T15_43_06_087Z-debug.log
```

I get the err above.

I double checked the above mentioned npmrc file and it has this content:

```
@radum:registry=https://registry.npmjs.org/
package-lock=false

//registry.npmjs.org/:_authToken = ${NPM_TOKEN}
```

Evaluating the `${NPM_TOKEN}` env var give me the correct automation token I have generated. I double checked this.

Even if I paste the token manually in that temp file I get the same err.

I think this is because npm whoami doesn't support automation tokens to be used. But the publish command will be ok with it.

Is my thinking correct here?

--- a/tweets/2020/npm-automation-tokens.tweet
+++ b/tweets/2020/npm-automation-tokens.tweet
@@ -1,0 +1,5 @@
+📯 semantic-release release now works with the new npm automation tokens. That means you can enable two-factor auth on your account, while still automate releases through CI.
+
+Thanks for confirming @grantila @hamzahamidii @thenieds
+
+https://github.blog/changelog/2020-10-02-npm-automation-tokens/

--- a/tweets/2020/npm-automation-tokens.tweet
+++ b/tweets/2020/npm-automation-tokens.tweet
@@ -1,4 +1,4 @@
-📯 semantic-release release now works with the new npm automation tokens. That means you can enable two-factor auth on your account, while still automate releases through CI.
+📯 semantic-release release now works with the new npm automation tokens. That means you can enable two-factor auth on your account, while still automating releases through CI.
 
 Thanks for confirming @grantila @hamzahamidii @thenieds
 

